analyses/upstream-analysis/util/function-process-signac.R

```diff
@@ -0,0 +1,322 @@
+#############################################################################################
+#' Function to process and cluster cell for an individual sample
+#' @param seurat_obj
+#' @param min.cutoff_value
+#' @param assay
+#' @param results_dir
+#' @param plots_output
+#' @param use_condition_split
+#' @param condition1
+#' @param condition2
+#' @param condition3
+#' @param print_pdf
+#' @param grouping
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' 
+process_signac <- function(seurat_obj, min.cutoff_value, assay, results_dir, 
+                           plots_output, use_condition_split, condition1, condition2, 
+                           condition3, print_pdf, grouping) {
+  
+  set.seed(1234) # Make code reproducible
+  
+  # Normalize data
+  seurat_obj <- RunTFIDF(seurat_obj)
+  seurat_obj <- FindTopFeatures(seurat_obj, min.cutoff = min.cutoff_value) 
+  seurat_obj <- RunSVD(seurat_obj) # SVD applied to term-document matrix, i.e., peak-call count matrix afer TF-IDF transformation; analogous to PCA
+  
+  # Plot DepthCor
+  #name <- paste0(plots_dir, "/", glue::glue("DepthCor.png"))
+  #print(DepthCor(seurat_obj))
+  #ggsave(file = name, width = 6, height = 5, device = "png")
+  
+  name <-  paste0(plots_dir, "/", glue::glue("DepthCor.pdf"))
+  p <- DepthCor(seurat_obj)
+  pdf(file = name, width = 6, height = 5)
+  print(p)
+  dev.off()
+  
+  cat("Run UMAP", "\n")
+  seurat_obj <- RunUMAP(seurat_obj, reduction = "lsi", dims = 2:30)
```

Comment:
Based on the tutorials linked, it seems like it quite common for the first dimension to be correlated with sequence depth and so needs to be dropped, but maybe there are rarer examples where that is not true? If there could be instance where this is not the case, perhaps you could consider making it an option to use `dims = 1:30` instead of `dims = 2:30`. For example, if the user reviews the `DepthCor` plot and determines dimension 1 should be used after all.